USS Western Eye - Alternate Shipmap

kugamo opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

![2023 03 03-20 40 59](https://user-images.githubusercontent.com/16618648/222872549-461289e7-cb4e-41c1-92a0-0509de30f279.jpg)
_theres alot of wonky mapping textures, it will look normal in game_

Full size image: https://media.discordapp.net/attachments/1076335601009299507/1081410753799659621/2023.03.03-20.40.59.png?width=792&height=528
# About the pull request

This PR adds a new shipmap into the game: the **USS Western Eye**. _(named with the help of your local lore team)_

Shipmap vote will occur at round end, every round. What you play on is up to you! 😨 

After playing the game for a number of years the problems with the Almayer had become evident to me, namely alot of bloated hallways maintenance areas, as well annoyingly long travel times, department size inconsistancies, cryopod locations spread out everywhere, LIFEBOATS REEE, and more that is hard to pick off the top of my head. I tried addressing all of them in the design of this new shipmap, heres a rough list of the major changes that might be noticed in comparison to the Almayer:

- Vertical orientation. Mostly done because I thought it would be interesting, but also allows for a third deck (The map size is the same as the Almayer's)
- No lifeboats, they're gone! In their stead I revived the SD (located on the third deck, triggered using communications console in CIC)
- All cryopods have been consolidated to a single location on the second deck (right by dorms), and as a result everyone will spawn together there primed to go and get some chow at the mess hall just fore of there.
- All officer gear vendors have been moved to Officer Prep on the third deck (right by the aft stairs for easy access)
- Evacuation pods have been segregated from the main hallways and shuttered off, and will only open when evac is called (can be opened from ARES room if need be).
- Expanded dorms area on the second deck, with bunks for each department. Marines during a prolonged active duty time now have somewhere to rest!
- Multi-zlevel cargo elevator. Completely new feature, travel between all the decks in style!


This has been a long time coming, huge thanks to @nauticall and @carlarctg for contributing to the map! Check out those maints!

I know its a huge PR with a multitude of changes (sorry). Much love reviewers <3 

*Also figuring out hijack impact location code, forgot about this.

# Explain why it's good for the game

New map, new fun.

# Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->
<!-- If you add a name after the ':cl', that name will be used in the changelog. You must add your CKEY after the CL if your GitHub name doesn't match. Be sure to properly mark your PRs to prevent unnecessary GBP loss. Maintainers freely reserve the right to remove and add tags should they deem it appropriate. -->

:cl:
add: Added functionality for multizlevel elevators.
fix: Fixed crushing cans not making noise, potted plants not selecting random sprites, and SD being out of date.
soundadd: Sounds for falling down zlevels, splat!
imageadd: Sprites for multizlevel elevator, transit walls, pit walls, freight door, and controller. Sprites for alternate landing zone lights directions. Sprites for the ship name, deck level signs.
mapadd: Added a new shipmap: USS Western Eye!
config: Added shuttles option to map configs, for map specific shuttle specification.
admin: Admins can now call shipmap votes.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! -->

kugamo committed:
Initial Commit

kugamo committed:
Part 2

Map layout generally finalized

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
part 3

not big change, just wanted to have them public

kugamo committed:
Basic Elevator working

This was hell

kugamo committed:
irons out most of the elevator bugs

just some minor stuff left

kugamo committed:
More Updates

kugamo committed:
More Progress

kugamo committed:
ELEVATOR FINISHED

more or less, ok new features always got some wrinkles ok, I spent way to long on this, ok, JUST ACCEPT IT

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
readding changes to admin Level

kugamo committed:
Latest Changes

kugamo committed:
Progress two shamnbiloouu

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
Rename + carlarcs contributions

kugamo committed:
Its about damn time

kugamo committed:
Vote Janks

kugamo committed:
Merge branch 'master' into Wayfinder

harryob commented on code/modules/shuttles/shuttle_supply.dm:
```diff
@@ -23,7 +23,7 @@
 /datum/shuttle/ferry/supply/proc/pick_loc()
 	RETURN_TYPE(/turf)
-	return GLOB.supply_elevator
+	return GLOB.supply_elevator_turfs[1]
```
check length of the list here - if it doesn't exist, trying to access the first one in the index will error (like in CI where there's no supply elevator for the groundmaps)

harryob replied:
```suggestion
    if(length(GLOB.supply_elevator_turfs))
		return GLOB.supply_elevator_turfs[1]
```
check length of the list here - if it doesn't exist, trying to access the first one in the index will error (like in CI where there's no supply elevator for the groundmaps)

hry-gh reviewed (changes requested):
you also need to repath all your security cameras to /cameras instead of /security

realforest2001 reviewed (changes requested):
![image](https://user-images.githubusercontent.com/41653574/219877004-0bb70e59-73ba-442c-bc86-deb5499cd817.png)

req_access_one_txt is currently "s"

realforest2001 commented:
Also I'm not sure what you've got going on in the top right of the map, but there's half an evac pod there causing an error

kugamo committed:
Replaces all the camera consoles

Hopefully :cow:

kugamo committed:
:cow:

kugamo committed:
:fire_engine:

kugamo committed:
Why does the linter check COMMENTS?!

YOU TELL ME

kugamo committed:
More funny proc references

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
:turtle:

harryob commented on code/modules/shuttles/escape_pods.dm:
```diff
@@ -132,6 +132,10 @@ for(var/obj/structure/machinery/cryopod/evacuation/C in cryo_cells) C.go_out()
 	evacuation_program.dock_state = STATE_LAUNCHING
 	spawn()
+		var/list/turf/Surrounding_turfs_to_D = list(get_step(D, NORTH), get_step(D, SOUTH), get_step(D, EAST), get_step(D, WEST))
+		for(var/i=1;i<=length(Surrounding_turfs_to_D);i++)
```
for(var/i in 1 to length(Surrounding_turfs_to_D))

harryob commented on code/modules/shuttles/escape_pods.dm:
```diff
@@ -132,6 +132,10 @@ for(var/obj/structure/machinery/cryopod/evacuation/C in cryo_cells) C.go_out()
 		return FALSE //Can't launch in some circumstances.
 	evacuation_program.dock_state = STATE_LAUNCHING
 	spawn()
+		var/list/turf/Surrounding_turfs_to_D = list(get_step(D, NORTH), get_step(D, SOUTH), get_step(D, EAST), get_step(D, WEST))
```
snake case (surrounding_turfs) and specify what D is actually meant to be

harryob commented on code/modules/shuttles/escape_pods.dm:
```diff
@@ -132,6 +132,10 @@ for(var/obj/structure/machinery/cryopod/evacuation/C in cryo_cells) C.go_out()
 	spawn()
+		var/list/turf/Surrounding_turfs_to_D = list(get_step(D, NORTH), get_step(D, SOUTH), get_step(D, EAST), get_step(D, WEST))
+		for(var/i=1;i<=length(Surrounding_turfs_to_D);i++)
+			for(var/obj/structure/machinery/door/poddoor/almayer/open/evac/Evac_Poddoor in Surrounding_turfs_to_D[i].contents)
```
snake case (evac_poddoor)

harryob commented on code/game/supplyshuttle.dm:
```diff
@@ -834,9 +876,40 @@ var/datum/controller/supply/supply_controller = new()
+						dat += "Moving<BR>"
+					if (shuttle.docking_controller)
+						switch(shuttle.docking_controller.get_docking_status())
+							if ("docked") dat += "Raised to level: [m_shuttle.fake_zlevel]<BR>"
+							if ("undocked") dat += "Lowered<BR>"
+							if ("docking") dat += "Moving to level [m_shuttle.fake_zlevel] [shuttle.can_force()? SPAN_WARNING("<A href='?src=\ref[src];force_send=1'>Force</A>") : ""]<BR>"
+							if ("undocking") dat += "Moving [shuttle.can_force()? SPAN_WARNING("<A href='?src=\ref[src];force_send=1'>Force</A>") : ""]<BR>"
```
no in line control statements
```suggestion
							if ("docked")
								dat += "Raised to level: [m_shuttle.fake_zlevel]<BR>"
							if ("undocked")
								dat += "Lowered<BR>"
							if ("docking")
								dat += "Moving to level [m_shuttle.fake_zlevel] [shuttle.can_force()? SPAN_WARNING("<A href='?src=\ref[src];force_send=1'>Force</A>") : ""]<BR>"
							if ("undocking")
								dat += "Moving [shuttle.can_force()? SPAN_WARNING("<A href='?src=\ref[src];force_send=1'>Force</A>") : ""]<BR>"
```

harryob commented on code/game/supplyshuttle.dm:
```diff
@@ -894,18 +965,44 @@ var/datum/controller/supply/supply_controller = new()
+		if(m_shuttle.moving_status == SHUTTLE_IDLE)
+			m_shuttle.target_zlevel =  m_shuttle.choices_to_number(tgui_input_list(usr, "Where should the shuttle move next?", "Pick Target", choices))
+		else
+			to_chat(usr, SPAN_WARNING("The Elevator cannot change targets while moving. Please wait."))
```
balloon alert here (they're cool)

```suggestion
			to_chat(usr, SPAN_WARNING("The Elevator cannot change targets while moving. Please wait."))
			usr.balloon_alert(usr, "not while moving!")
```

kugamo replied:
ok bet

harryob commented on code/game/turfs/walls/wall_types.dm:
```diff
@@ -65,6 +65,53 @@
+
+	var/list/turf_neighbors = list(get_step(src, NORTH).density, get_step(src, SOUTH).density, get_step(src, EAST).density, get_step(src, WEST).density)
+	var/number_of_neighbors
+	for(var/i=1,i<=length(turf_neighbors),i++)
```
```suggestion
	for(var/i in 1 to length(turf_neighbors)
```

harryob commented on code/game/turfs/floor_types.dm:
```diff
@@ -264,6 +265,67 @@
+			var/mob/living/carbon/human/human_who_fell = AM
+			human_who_fell.take_overall_damage(rand(40, 60), 0, FALSE, FALSE, "Blunt Trauma")
+			human_who_fell.KnockDown(5)
+			for(var/i in human_who_fell.limbs)
+				if(istype(i, /obj/limb/leg/l_leg/) || istype(i, /obj/limb/leg/r_leg))
+					var/obj/limb/leg/human_who_fells_leg = i
+					human_who_fells_leg.take_damage_bone_break(35)
```
test this, but

```suggestion
			for(var/obj/limb/leg in human_who_fell.limbs)
				leg.take_damage_bone_break(35)
```

kugamo replied:
and its GOoooD! :touchdown:

harryob commented on code/game/turfs/floor_types.dm:
```diff
@@ -264,6 +265,67 @@
+					AM.layer = UNDER_TURF_LAYER
+					break
+			else if(my_area.fake_zlevel + 1 <= length(GLOB.supply_elevator_turfs))
+				for(var/i=0; i<101; i++)
```
```suggestion
				for(var/i in 0 to 101)
```
what's this magic number also

kugamo replied:
line 279, I thought about making this number part of the map config... but ya know

kugamo replied:
In next commit I added a proc that finds this number based on searching the map for it, so magic number solved.

harryob commented on code/game/objects/items/reagent_containers/food/cans.dm:
```diff
@@ -115,41 +122,41 @@
 
+		var/mob/living/carbon/human/H = user
+		var/message
+		var/obj/limb/L
+		L = H.get_limb(H.zone_selected)
```
```suggestion
		var/obj/limb/L = H.get_limb(H.zone_selected)
```

harryob commented on code/game/objects/items/reagent_containers/food/cans.dm:
```diff
@@ -115,41 +122,41 @@
+		L = H.get_limb(H.zone_selected)
+
+		if(src == H.get_inactive_hand())
+			message = "between [user.gender == MALE ? "his" : "her"] hands"
```
```suggestion
			message = "between [user.p_their()] hands"
```

harryob commented on code/game/objects/items/reagent_containers/food/cans.dm:
```diff
@@ -115,41 +122,41 @@
+					if(!L)
+						to_chat(user, SPAN_WARNING("You don't have a [H.zone_selected], can't crush yer can under nothing!"))
+						return
+					message = "under [user.gender == MALE ? "his" : "her"] foot!"
```
```suggestion
					message = "under [p_their()] foot!"
```

harryob commented on code/game/objects/items/reagent_containers/food/cans.dm:
```diff
@@ -115,41 +122,41 @@
+					if(!L)
+						to_chat(user, SPAN_WARNING("You don't have a [H.zone_selected], can't crush yer can on nothing!"))
+						return
+					message = "against [user.gender == MALE ? "his" : "her"] head!"
```
```suggestion
					message = "against [user.p_their()] head!"
```

harryob commented on code/game/objects/items/reagent_containers/food/cans.dm:
```diff
@@ -4,6 +4,13 @@
+
+	if(crushed) //for my mapper friens
+		reagents.clear_reagents()
+		crush_can(null, TRUE)
```
```suggestion
		crush_can(override_for_map_var_edits = TRUE)
```
null args ugly!!

harryob commented on code/game/objects/effects/projector.dm:
```diff
@@ -8,3 +8,38 @@
 	icon = 'icons/landmarks.dmi'
 	icon_state = "projector"//for map editor
+	var/layer_override
+	var/paused = FALSE // 0 for not paused, 1 for primed for pausaztion, 2 for completely paused
+	var/list/coord_cache //used to return after moving
```
```suggestion
	/// 0 for not paused, 1 for primed for pausaztion, 2 for completely paused
	var/paused = FALSE
	///used to return after moving
	var/list/coord_cache
```

harryob commented on code/game/objects/effects/elevator.dm:
```diff
@@ -20,5 +20,20 @@
 
 /obj/effect/elevator/animation_overlay
-	blend_mode = BLEND_INSET_OVERLAY
+	//blend_mode = BLEND_INSET_OVERLAY
```
? - remove if unused

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+/datum/shuttle/ferry/supply/multi
+	iselevator = 1
+	var/list/obj/effect/elevator/supply/multi/elevator_effects = list()
+	var/list/shaft_projectors = list() //contains lists which each contain groups of projectors near each stop of the elevator
```
```suggestion
	/// contains lists which each contain groups of projectors near each stop of the elevator
	var/list/shaft_projectors = list()
```

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+	var/obj/structure/elevator_control_mount/in_elevator_controller
+	replacement_turf_type = /turf/open/floor/almayer/empty/multi_elevator
+	var/list/transit_location_walls = list()
+	var/area/shaft_transit_area  //purposefully defined this as something other than transit_area to not trigger shuttle code
+	var/list/obj/effect/elevator/animation_overlay/elevator_animationz = list()
+	var/list/railingz = list() //z for coolness
```
```suggestion
	/// purposefully defined this as something other than transit_area to not trigger shuttle code
	var/area/shaft_transit_area  
	var/list/obj/effect/elevator/animation_overlay/elevator_animationz = list()
	/// z for coolness
	var/list/railingz = list() 
```

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+	warmup_time = 5 SECONDS
+
+/datum/shuttle/ferry/supply/multi/New()
+	for(var/i = 1; i <= length(GLOB.supply_elevator_turfs); i++)
```
```suggestion
	for(var/i in 1 to length(GLOB.supply_elevator_turfs))
```

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+	for(var/turf/T in GLOB.supply_elevator_turfs)
+		var/area/T_area = get_area(T)
+		if(length(railingz) < length(GLOB.supply_elevator_turfs))
+			for(var/i=1;i < length(GLOB.supply_elevator_turfs); i++)
```
```suggestion
			for(var/i in 1 tolength(GLOB.supply_elevator_turfs))
```

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+	for(var/obj/effect/projector/P in projectors)
+		for(var/turf/T in GLOB.supply_elevator_turfs)
+			var/area/area_of_T = get_area(T)
+			for(var/i = 0; i < area_of_T.fake_zlevel; i++)
```
```suggestion
			for(var/i in 0 to area_of_T.fake_zlevel)
```

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+						shaft_projectors[area_of_T.fake_zlevel] |= P
+
+	if(supply_controller)
+		for(var/i=1; i <= length(GLOB.supply_elevator_turfs);i++)
```
```suggestion
		for(var/i in 1 to length(GLOB.supply_elevator_turfs))
```

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+		return shaft_transit_area
+
+/datum/shuttle/ferry/supply/multi/proc/get_movement_direction()
+	//0 means DOWN; 1 means UP
```
use defines for this

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+	if(isnull(location) || target_zlevel == fake_zlevel)
+		return
+
+	recharging = 1
```
is this a bool? use TRUE

kugamo replied:
it aint, its rounded in long_jump()??? shuttle code jank

harryob replied:
what the hell

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+	move_elevator_to_transit(origin, shaft_transit_area)
+
+	var/timing_multiplier = (target_zlevel > fake_zlevel) ? target_zlevel - fake_zlevel : fake_zlevel - target_zlevel
+	sleep(5 SECONDS * timing_multiplier) //muh transit time
```
timer 🥺 in this case it's just calling move_elevator_to_destination after 5 SECONDS anyway

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+
+	move_elevator_to_destination(shaft_transit_area, destination)
+
+	spawn(2 SECONDS) //animations need to catch up
```
timer 🥺 just throw this stuff into a throwaway proc and call it after 2 seconds

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+		var/at_station_result = at_station()
+		if (!at_station_result)	//at centcom
+			handle_sell()
+		spawn(0)
```
what this for

kugamo replied:
idk old copy paste shuttle shitcode

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+	if(!deck)
+		return
+
+	var/effective = 0
```
if this is a bool use FALSE

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+		elevator_animationz[target_zlevel].color = "#577eaa"
+		animate(elevator_animationz[target_zlevel], pixel_y = 0, color = "#ffffff", time = 2 SECONDS)
+
+	spawn(2 SECONDS)
```
bestie could you...

for reference it should be something like !!!! pseudocode !!!!
```dm
addtimer(CALLBACK(src, PROC_REF(do_move_up), origin, destination, target_zlevel), 2 SECONDS)

/datum/shuttle/ferry/supply/multi/proc/do_move_up(origin, destination, target_zlevel)
	move(origin, destination)
 	elevator_animationz[target_zlevel].vis_contents.Cut()

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+	//and that lower zlevels are ordered from 1 up moving downwards
+	//all logic handled by /datum/controller/subsystem/fz_transitions/fire()
+	switch(mode)
+		if(1)
```
defines

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+						P.update_state(FALSE)
+
+/datum/shuttle/ferry/supply/multi/forbidden_atoms_check()
+	if(target_zlevel != 0 || target_zlevel != null)
```
```suggestion
	if(target_zlevel)
```

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+			//in_elevator_controller.attached_to.choices[fake_zlevel].underlays += icon('icons/obj/items/misc.dmi', "ec_button_current")
+
+/datum/shuttle/ferry/supply/multi/proc/choices_to_number(choice)
+	//ha ha snowflake code here made to only handle 3 decks *dabs*
```
it's so over for you

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+
+/datum/shuttle/ferry/supply/multi/proc/choices_to_number(choice)
+	//ha ha snowflake code here made to only handle 3 decks *dabs*
+	switch(choice)
+		if("ASRS")
+			return 0
+		if("Deck 1")
+			return 1
+		if("Deck 2")
+			return 2
+		if("Deck 3")
+			return 3
+		if(0)
+			return "ASRS"
+		if(1)
+			return "Deck 1"
+		if(2)
+			return "Deck 2"
+		if(3)
+			return "Deck 3"
```
!!!! pseudocode here !!!!

```dm
	var/static/list/decks = list(
		0 = "ASRS",
		1 = "Deck 1",
		2 = "Deck 2",
		3 = "Deck 3",
	)
	
	if(decks[choices])
		return decks[choices]
		
	for(var/level in decks)
		if(decks[level] == choices)
			return level
```

harryob commented on code/modules/shuttles/shuttle_supply_multi.dm:
```diff
@@ -0,0 +1,825 @@
+
+	update_icon()
+
+/obj/structure/elevator_control_mount/proc/override_delete()
+	SIGNAL_HANDLER
+	recall_phone()
+	return COMPONENT_ABORT_QDEL
```
```suggestion
/obj/structure/elevator_control_mount/proc/override_delete(force)
	SIGNAL_HANDLER
	recall_phone()
	if(!force)
		return COMPONENT_ABORT_QDEL
```
this whole thing probably needs reworking (the phone code isn't that good, ik that's where you got it from) but uhh dw about it unless you really feel like it

(the phone should be made ephemeral and spawn in when you attackhand the phone, and despawn when you put it back down)

hry-gh reviewed (changes requested):
let's gooooooooo

kugamo commented on code/game/turfs/floor_types.dm:
```diff
@@ -264,6 +265,67 @@
+	if(AM.throwing == 0 && istype(get_turf(AM), type))
+		//This is assuming the shaft is alined across fake_zlevels -- if you're getting moved to random locations blame the mappers
+		//also that the fake zlevels of the shipmap are 101 tiles apart
+		message_admins("[AM] ---> [AM.type]")
```
yea

kugamo committed:
Harry's review + Dorms redux

CallmeArty commented:
Your ship is straight sexy as hell, but you're missing some (literal) signposting for helping people navigate such a spacehulk. Like Agameofscones said, some distinction is necessary.

Baystation's SEV Torch - [interactive map here](https://affectedarc07.github.io/SS13WebMap/Bay12/Torch/) - is best-in-class for this, in my humble HRP opinion: Each deck has a color-coordinated lining along the walls that denotes the purpose of the area (green for the recreation deck with a red trim around the security booth), in addition to directional sign that point you towards other departments. If I got shot in briefing, there are no visual indicators (MED ->, lines on the floor, etc) of where I'm supposed to go unless I already know.

Medmain input: This is not a medic-condusive lobby/hangar design: Both an approach from squad prep (pink) and briefing (blue) - which are in my experience, the most common times for a medic to drop-by for chems - require crossing over another path of traffic, and looping back around. [Standard marine flow in yellow for comparison].
![image](https://user-images.githubusercontent.com/64385385/220464990-92870122-56d4-477c-b457-fe2ebfe919e3.png)

The comparatively minor change to help with this would simply be swapping the positions of the chemlab entrance, and it's gates/fridge, allowing for medics coming from prep (red) to easily keep one path from prep, to chem, to briefing or the dropships, or at least a briefing-chem-ship (green) route that wouldn't take them infront of the medbay doors.
![image](https://user-images.githubusercontent.com/64385385/220466555-61c20571-e26b-4d6f-8bd5-8e60b3300e1f.png)

kugamo commented:
> Your ship is straight sexy as hell, but you're missing some (literal) signposting for helping people navigate such a spacehulk. Like Agameofscones said, some distinction is necessary.
> 
> Baystation's SEV Torch - [interactive map here](https://affectedarc07.github.io/SS13WebMap/Bay12/Torch/) - is best-in-class for this, in my humble HRP opinion: Each deck has a color-coordinated lining along the walls that denotes the purpose of the area (green for the recreation deck with a red trim around the security booth), in addition to directional sign that point you towards other departments. If I got shot in briefing, there are no visual indicators (MED ->, lines on the floor, etc) of where I'm supposed to go unless I already know.
> 
> Medmain input: This is not a medic-condusive lobby/hangar design: Both an approach from squad prep (pink) and briefing (blue) - which are in my experience, the most common times for a medic to drop-by for chems - require crossing over another path of traffic, and looping back around. [Standard marine flow in yellow for comparison]. ![image](https://user-images.githubusercontent.com/64385385/220464990-92870122-56d4-477c-b457-fe2ebfe919e3.png)
> 
> The comparatively minor change to help with this would simply be swapping the positions of the chemlab entrance, and it's gates/fridge, allowing for medics coming from prep (red) to easily keep one path from prep, to chem, to briefing or the dropships, or at least a briefing-chem-ship (green) route that wouldn't take them infront of the medbay doors. ![image](https://user-images.githubusercontent.com/64385385/220466555-61c20571-e26b-4d6f-8bd5-8e60b3300e1f.png)

All these lines are very wrong. Prep is on the third deck yet you have the "most common path of traffic" coming up from the third deck. The quickest route is through the ladders fore of briefing, which you neglected to show on your diagrams. Anyways if medbay is alittle out of the way I don't think thats a terrible thing. 

I'll expirament with the departmental borders, you're right they're alittle sparse on my map.

kugamo committed:
MANY THINGS

->Animations for where the falling thing through zlevel goes (not just for leaving the area they came from)
->Hijack location picking is specific now and supports multiple ship maps
-> fixing some  code

kugamo committed:
squashing da BUGz

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
Readding Conflicted files' changes

kugamo committed:
Bryuh

CallmeArty commented:
> All these lines are very wrong. Squad's preps are on the firstdeck yet you have the "most common path of traffic" coming up from the third deck. The quickest route to medbay from prep/briefing is actually through the ladders fore of briefing, which you neglected to show on your diagrams. **Anyways if medbay is alittle out of the way I don't think thats a terrible thing.**
> 
> I'll experiment with the departmental borders, you're right they're alittle sparse on my map.

My apologies, as I was attempting to eyeball the linked staircases, and had assumed those ladders fore of briefing were inaccessible to marines as the similar ones (security checkpoints) on the Almayer are.

Presently, there isn't much spare time in rounds, as current attitudes and metas tend to try to minimize time out of cryo, and into deployment. For that reason and assuming attitudes don't shift: trying to align as straight a shot as possible for marines to wake-up, get geared, brief, and leave would be ideal.

Some decisions like having research and the morgue on the same deck as the dropships - the two biggest corpse-dumps on a server/gamemode which is a corpse factory - are terrific for QOL and gameflow; and the ordinance workshop being close enough to deal to prepping marines and req. However, I have spent a _lot_ of time as medic or CMO, and can say that it is a common sight for squads to either drop without medics, or with under-supplied medics due to any time loss (latejoin, long line in requisition, slow doctors, etc), and it can have a noticeable impact on rounds.

kugamo committed:
TAGS BE GONE

also notes

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
EVAC FIX

kugamo committed:
Removes vendors with rubber bullets from Firing range

my bad

XanHiroki commented:
Would like to see a squad distribution computer in the SEA's office, a lot of mentors would agree.

kugamo committed:
HOly Moly

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
fixes mostly everything but elevator

kugamo committed:
Wahoo Letsago

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
yep

kugamo committed:
Tags!?

kugamo committed:
Brooo Bruh boo

kugamo committed:
:skull:

kugamo committed:
TAGS AGAIN

kugamo committed:
A window in my wall? I dont think so

:nose:

kugamo committed:
Finally finishes elevators

for real this time no cap on god

kugamo committed:
:house_with_garden:

kugamo committed:
skull emoji

:skull:

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
:cancer:

kugamo committed:
SD layout rework

kugamo committed:
req redesign

RenaRenaRe commented:
OT's requisition delivery unit does not work. You click engage handle and it just shoots the contents back out at you.
Also there's no mortar camera monitor.

Weuyn commented:
Same thing with the research delivery chute, but that was always broken. I think disposals units in general are broken and rarely flush properly.

Research is missing the 6 or 8 vial boxes it starts with on the almayer. You need those for programming the turing, running the centrifuge and distributing autoinjectors.

kugamo committed:
Shinks the map

 to be under 256 in dimensions

kugamo committed:
Fixes TM issues

kugamo committed:
operatingBgone

kugamo committed:
Wowowowoowow

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
ohohoho

cm13-github commented:
#### Test Merge Deployed

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: a6615543269deafdf5ad1f04caa9db43ed653967
Pull Request: a3f9949fec8ef0b879fb37e5c4c4bd06f6a19abc
Server: 566e2a7569e3c8fe4e83ec05f9b162d1c50e20d2

##### Comment

</details>

cm13-github commented:
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: a6615543269deafdf5ad1f04caa9db43ed653967
Pull Request: a3f9949fec8ef0b879fb37e5c4c4bd06f6a19abc
Server: e03f29db11392a076bad2f02448b2f3a3ac49415

##### Comment

</details>

cm13-github commented:
#### Test Merge Removed

kugamo committed:
most of the fixes

animations are weird on the elevator for some reason

Herotitan772 commented:
Staff members should have vendors in their preps like on the alamyer. Referring to the CE, RO, XO, ect

also people should spawn in their relevant departments on round start, eg. CE in their office, MTs in engineering, doctors in Medbay, pilots in their office, ect

kugamo commented:
> Staff members should have vendors in their preps like on the alamyer. Referring to the CE, RO, XO, ect
> 
> also people should spawn in their relevant departments on round start, eg. CE in their office, MTs in engineering, doctors in Medbay, pilots in their office, ect

no

cm13live commented:
This pull request has been mentioned on **CM-SS13**. There might be relevant details there:

https://forum.cm-ss13.com/t/2676-uss-western-eye-alternate-shipmap/812/1

kugamo committed:
Merge branch 'master' into Wayfinder

kugamo committed:
OT & MessTech now have spawns!

Googles-Hands commented:
What a shame, I would have liked to see the multi-z ASRS.

Drulikar commented:
Since it was poorly communicated: Kugamo explicitly did not want this project to continue as of ~04/26/2023

realforest2001 commented:
As of 06/25/2026 permission has been granted for continuation.

Thegenearl commented:
Holy peak?
